config.org
#+begin_src elisp :tangle yes

;;; $DOOMDIR/config.el -*- lexical-binding: t; -*-

;; Place your private configuration here! Remember, you do not need to run 'doom
;; sync' after modifying this file!


;; Some functionality uses this to identify you, e.g. GPG configuration, email
;; clients, file templates and snippets.
(setq user-full-name "Thomas F. K. Jorna"
      user-mail-address "hello@tefkah.com")

;; Doom exposes five (optional) variables for controlling fonts in Doom. Here
;; are the three important ones:
;;
;; + `doom-font'
;; + `doom-variable-pitch-font'
;; + `doom-big-font' -- used for `doom-big-font-mode'; use this for
;;   presentations or streaming.
;;
;; They all accept either a font-spec, font string ("Input Mono-12"), or xlfd
;; font string. You generally only need these two:
;; (setq doom-font (font-spec :family "monospace" :size 12 :weight 'semi-light)
;;       doom-variable-pitch-font (font-spec :family "sans" :size 13))

;; There are two ways to load a theme. Both assume the theme is installed and
;; available. You can either set `doom-theme' or manually load a theme with the
;; `load-theme' function. This is the default:
(setq doom-theme 'doom-one)

;; If you use `org' and don't want your org files in the default location below,
;; change `org-directory'. It must be set before org loads!
(setq org-directory "~/Notes/")

;; This determines the style of line numbers in effect. If set to `nil', line
;; numbers are disabled. For relative line numbers, set this to `relative'.
(setq display-line-numbers-type t)


;; Here are some additional functions/macros that could help you configure Doom:
;;
;; - `load!' for loading external *.el files relative to this one
;; - `use-package!' for configuring packages
;; - `after!' for running code after a package has loaded
;; - `add-load-path!' for adding directories to the `load-path', relative to
;;   this file. Emacs searches the `load-path' when you load packages with
;;   `require' or `use-package'.
;; - `map!' for binding new keys
;;
;; To get information about any of these functions/macros, move the cursor over
;; the highlighted symbol at press 'K' (non-evil users must press 'C-c c k').
;; This will open documentation for it, including demos of how they are used.
;;
;; You can also try 'gd' (or 'C-c c d') to jump to their definition and see how
;; they are implemented.
#+end_src 

#+begin_src emacs-lisp :tangle yes
(evil-workman-global-mode 1)
#+end_src


It would be nice to see the characters when switching windows.

#+BEGIN_SRC elisp :tangle yes
(custom-set-faces!
  '(aw-leading-char-face
    :foreground "white" :background (face-background 'hl-line)
    :weight bold :height 2.5 :box (:line-width 10 :color "red")))
#+END_SRC

This makes treemacs better or something

#+begin_src elisp :tangle yes
(setq treemacs-git-mode 'extended)
#+end_src

Boggles the mind that we use monospace for everything in emacs. Luckily someone smarter than me put in some time to rectify this.

This activates fixed pitch mode, which turns all "text" to variable pitch.

#+begin_src emacs-lisp :tangle yes
(setq doom-font "FiraCode Nerd Font")
(setq doom-variable-pitch-font "Roboto")
(setq default-frame-alist '((font . "Roboto")
                             (internal-border-width 12)
                             (vertical-scroll-bars nil)))
#+end_src

Love my little floating man

#+begin_src emacs-lisp :tangle yes
(vertico-posframe-mode 1)
#+end_src

Make ~which-key~ a bit faster

#+begin_src emacs-lisp :tangle yes
(setq which-key-idle-delay 0.4)
#+end_src

Remove ~evil~ prefix from a lot of things (thanks tecosaur 🙂)

#+begin_src emacs-lisp :tangle yes
(setq which-key-allow-multiple-replacements t)
(after! which-key
  (pushnew!
   which-key-replacement-alist
   '(("" . "\\`+?evil[-:]?\\(?:a-\\)?\\(.*\\)") . (nil . "◂\\1"))
   '(("\\`g s" . "\\`evilem--?motion-\\(.*\\)") . (nil . "◃\\1"))
   ))
#+end_src


** Info colors

#+begin_src emacs-lisp :tangle yes
(add-hook 'Info-selection-hook 'info-colors-fontify-mode)
#+end_src

** Marginalia

Once again, thanks teco

#+begin_src emacs-lisp :tangle yes
(after! marginalia
  (setq marginalia-censor-variables nil)

  (defadvice! +marginalia--anotate-local-file-colorful (cand)
    "Just a more colourful version of `marginalia--anotate-local-file'."
    :override #'marginalia--annotate-local-file
    (when-let (attrs (file-attributes (substitute-in-file-name
                                       (marginalia--full-candidate cand))
                                      'integer))
      (marginalia--fields
       ((marginalia--file-owner attrs)
        :width 12 :face 'marginalia-file-owner)
       ((marginalia--file-modes attrs))
       ((+marginalia-file-size-colorful (file-attribute-size attrs))
        :width 7)
       ((+marginalia--time-colorful (file-attribute-modification-time attrs))
        :width 12))))

  (defun +marginalia--time-colorful (time)
    (let* ((seconds (float-time (time-subtract (current-time) time)))
           (color (doom-blend
                   (face-attribute 'marginalia-date :foreground nil t)
                   (face-attribute 'marginalia-documentation :foreground nil t)
                   (/ 1.0 (log (+ 3 (/ (+ 1 seconds) 345600.0)))))))
      ;; 1 - log(3 + 1/(days + 1)) % grey
      (propertize (marginalia--time time) 'face (list :foreground color))))

  (defun +marginalia-file-size-colorful (size)
    (let* ((size-index (/ (log10 (+ 1 size)) 7.0))
           (color (if (< size-index 10000000) ; 10m
                      (doom-blend 'orange 'green size-index)
                    (doom-blend 'red 'orange (- size-index 1)))))
      (propertize (file-size-human-readable size) 'face (list :foreground color)))))
#+end_src


** Centaur tabs shenanigans

#+begin_src emacs-lisp :tangle yes
(after! centaur-tabs
  (centaur-tabs-mode -1)
  (setq centaur-tabs-height 36
        centaur-tabs-set-icons t
        centaur-tabs-modified-marker "o"
        centaur-tabs-close-button "×"
        centaur-tabs-set-bar 'above
        centaur-tabs-gray-out-icons 'buffer)
  (centaur-tabs-change-fonts "Roboto" 120))
#+end_src


** Olivetti mode

My love

It's so good, but make sure to use the fringes for the margins!

#+begin_src emacs-lisp :tangle yes
;(setq olivetti-style 'fancy)
#+end_src

But doom sets the colors of the fringes to a solair one, needs to be fixed.

#+begin_src emacs-lisp :tangle yes
(add-hook! 'solaire-mode-hook
  (set-face-attribute 'fringe nil :background (face-background 'solaire-default-face)))
#+end_src


* Org-Mode

Org-mode is pretty cool, but I have a loooooooot of small things I would like to have different.

** Small tweaks

*** Company mode

I'm not a big fan of company mode in org-mode, I've never wanted to have a word autocomplete when I'm typing prose, certainly not the random words that happen when typing in org-mode.
Also it will autocomplete words when typing in code blocks, which is rarely helpful.

#+begin_src emacs-lisp :tangle yes
(add-hook! org-mode-hook (company-mode -1))
#+end_src

** Org-Roam

#+begin_src emacs-lisp :tangle yes
(setq org-roam-directory "~/Notes")
(setq org-directory "~/Notes")
#+end_src

*** Org-roam-ui

My bb.
It's of course perfectly configured already so there's nothing to change.
#+begin_src emacs-lisp :tangle yes
(use-package! org-roam-ui
  :after org-roam)
#+end_src

* Looks

** General
*** Theme

#+begin_src emacs-lisp :tangle yes
(setq doom-theme 'doom-flatwhite)
#+end_src

*** Olivetti

I would like most non-prog buffers to have some more spacing.

#+begin_src emacs-lisp :tangle yes
(use-package! olivetti
  :hook (text-mode . olivetti-mode)
  :config)

#+end_src

*** Tooltips

Emacs actually has built in tooltips! You can just hover over things and then see what it means etc, wayyy more useful than the lower left corner.

#+begin_src emacs-lisp :tangle yes
(tooltip-mode 1)
#+end_src

** Org mode

*** Indentation

No indentation! It's ass.
#+begin_src emacs-lisp :tangle yes
(after! org
  (setq org-startup-indented nil))
#+end_src

*** Latex

I want the LaTeX previews to be there when I enter a file.
#+begin_src emacs-lisp :tangle yes
(after! org (setq org-startup-with-latex-preview t))
#+end_src

*** Org modern

Finally a standardized nice look for org

#+begin_src emacs-lisp :tangle yes
(use-package! org-modern
:hook (org-mode . org-modern-mode)
:config (setq org-modern-hide-stars t))
#+end_src

*** Mixed pitch

This is text to be read, it shouldn't be in fixed-pitch! Looks like ass!
Sadly, mixed-pitch-mode does not seem to change this automatically.


#+begin_src emacs-lisp :tangle yes
(add-hook! 'org-mode-hook (mixed-pitch-mode 1))
#+end_src

*** Olivetti

I want chapters to be centered.

#+begin_src emacs-lisp :tangle yes
(defun thomas/check-tag (tag)
  "Check whether current file-level note has a certain tag."
  (when (org-roam-file-p)
  (let* ((file-level-node
          (save-excursion
            (org-with-wide-buffer
            (buffer-end -1)
            (org-roam-node-at-point))))
         (file-tags (org-roam-node-tags file-level-node)))
    (if (seq-contains-p file-tags tag)
        t
      nil))))

(defun thomas/check-chapter-note ()
  "Check if current note is a chapter."
  (thomas/check-tag "chapter"))

(defun thomas/olivetti-if-chapter ()
  "Activate olitevetti mode if the current note is a chapter."
  (when (and (thomas/check-chapter-note))
    (setq-local olivetti-style 'fancy)
    (olivetti-mode 1)))


#+end_src

#+begin_src emacs-lisp :tangle yes
(add-hook 'org-mode-hook #'thomas/olivetti-if-chapter)
#+end_src



*** Line-numbers

No line numbers! I'm writing, I don't need to know where things are.

#+begin_src emacs-lisp :tangle yes
(add-hook! 'org-mode-hook (display-line-numbers-mode -1))
#+end_src


config.el
;;; $DOOMDIR/config.el -*- lexical-binding: t; -*-

;; Place your private configuration here! Remember, you do not need to run 'doom
;; sync' after modifying this file!


;; Some functionality uses this to identify you, e.g. GPG configuration, email
;; clients, file templates and snippets.
(setq user-full-name "Thomas F. K. Jorna"
      user-mail-address "hello@tefkah.com")

;; Doom exposes five (optional) variables for controlling fonts in Doom. Here
;; are the three important ones:
;;
;; + `doom-font'
;; + `doom-variable-pitch-font'
;; + `doom-big-font' -- used for `doom-big-font-mode'; use this for
;;   presentations or streaming.
;;
;; They all accept either a font-spec, font string ("Input Mono-12"), or xlfd
;; font string. You generally only need these two:
;; (setq doom-font (font-spec :family "monospace" :size 12 :weight 'semi-light)
;;       doom-variable-pitch-font (font-spec :family "sans" :size 13))

;; There are two ways to load a theme. Both assume the theme is installed and
;; available. You can either set `doom-theme' or manually load a theme with the
;; `load-theme' function. This is the default:
(setq doom-theme 'doom-one)

;; If you use `org' and don't want your org files in the default location below,
;; change `org-directory'. It must be set before org loads!
(setq org-directory "~/Notes/")

;; This determines the style of line numbers in effect. If set to `nil', line
;; numbers are disabled. For relative line numbers, set this to `relative'.
(setq display-line-numbers-type t)


;; Here are some additional functions/macros that could help you configure Doom:
;;
;; - `load!' for loading external *.el files relative to this one
;; - `use-package!' for configuring packages
;; - `after!' for running code after a package has loaded
;; - `add-load-path!' for adding directories to the `load-path', relative to
;;   this file. Emacs searches the `load-path' when you load packages with
;;   `require' or `use-package'.
;; - `map!' for binding new keys
;;
;; To get information about any of these functions/macros, move the cursor over
;; the highlighted symbol at press 'K' (non-evil users must press 'C-c c k').
;; This will open documentation for it, including demos of how they are used.
;;
;; You can also try 'gd' (or 'C-c c d') to jump to their definition and see how
;; they are implemented.

(evil-workman-global-mode 1)

(custom-set-faces!
  '(aw-leading-char-face
    :foreground "white" :background (face-background 'hl-line)
    :weight bold :height 2.5 :box (:line-width 10 :color "red")))

(setq treemacs-git-mode 'extended)

(setq doom-font "FiraCode Nerd Font")
(setq doom-variable-pitch-font "Roboto")
(setq default-frame-alist '((font . "Roboto")
                             (internal-border-width 12)
                             (vertical-scroll-bars nil)))

(vertico-posframe-mode 1)

(setq which-key-idle-delay 0.4)

(setq which-key-allow-multiple-replacements t)
(after! which-key
  (pushnew!
   which-key-replacement-alist
   '(("" . "\\`+?evil[-:]?\\(?:a-\\)?\\(.*\\)") . (nil . "◂\\1"))
   '(("\\`g s" . "\\`evilem--?motion-\\(.*\\)") . (nil . "◃\\1"))
   ))

(add-hook 'Info-selection-hook 'info-colors-fontify-mode)

(after! marginalia
  (setq marginalia-censor-variables nil)

  (defadvice! +marginalia--anotate-local-file-colorful (cand)
    "Just a more colourful version of `marginalia--anotate-local-file'."
    :override #'marginalia--annotate-local-file
    (when-let (attrs (file-attributes (substitute-in-file-name
                                       (marginalia--full-candidate cand))
                                      'integer))
      (marginalia--fields
       ((marginalia--file-owner attrs)
        :width 12 :face 'marginalia-file-owner)
       ((marginalia--file-modes attrs))
       ((+marginalia-file-size-colorful (file-attribute-size attrs))
        :width 7)
       ((+marginalia--time-colorful (file-attribute-modification-time attrs))
        :width 12))))

  (defun +marginalia--time-colorful (time)
    (let* ((seconds (float-time (time-subtract (current-time) time)))
           (color (doom-blend
                   (face-attribute 'marginalia-date :foreground nil t)
                   (face-attribute 'marginalia-documentation :foreground nil t)
                   (/ 1.0 (log (+ 3 (/ (+ 1 seconds) 345600.0)))))))
      ;; 1 - log(3 + 1/(days + 1)) % grey
      (propertize (marginalia--time time) 'face (list :foreground color))))

  (defun +marginalia-file-size-colorful (size)
    (let* ((size-index (/ (log10 (+ 1 size)) 7.0))
           (color (if (< size-index 10000000) ; 10m
                      (doom-blend 'orange 'green size-index)
                    (doom-blend 'red 'orange (- size-index 1)))))
      (propertize (file-size-human-readable size) 'face (list :foreground color)))))

(after! centaur-tabs
  (centaur-tabs-mode -1)
  (setq centaur-tabs-height 36
        centaur-tabs-set-icons t
        centaur-tabs-modified-marker "o"
        centaur-tabs-close-button "×"
        centaur-tabs-set-bar 'above
        centaur-tabs-gray-out-icons 'buffer)
  (centaur-tabs-change-fonts "Roboto" 120))

;(setq olivetti-style 'fancy)

(add-hook! 'solaire-mode-hook
  (set-face-attribute 'fringe nil :background (face-background 'solaire-default-face)))

(add-hook! org-mode-hook (company-mode -1))

(setq org-roam-directory "~/Notes")
(setq org-directory "~/Notes")

(use-package! org-roam-ui
  :after org-roam)

(setq doom-theme 'doom-flatwhite)

(use-package! olivetti
  :hook (text-mode . olivetti-mode)
  :config)

(tooltip-mode 1)

(after! org
  (setq org-startup-indented nil))

(after! org (setq org-startup-with-latex-preview t))

(use-package! org-modern
:hook (org-mode . org-modern-mode)
:config (setq org-modern-hide-stars t))

(add-hook! 'org-mode-hook (mixed-pitch-mode 1))

(defun thomas/check-tag (tag)
  "Check whether current file-level note has a certain tag."
  (when (org-roam-file-p)
  (let* ((file-level-node
          (save-excursion
            (org-with-wide-buffer
            (buffer-end -1)
            (org-roam-node-at-point))))
         (file-tags (org-roam-node-tags file-level-node)))
    (if (seq-contains-p file-tags tag)
        t
      nil))))

(defun thomas/check-chapter-note ()
  "Check if current note is a chapter."
  (thomas/check-tag "chapter"))

(defun thomas/olivetti-if-chapter ()
  "Activate olitevetti mode if the current note is a chapter."
  (when (and (thomas/check-chapter-note))
    (setq-local olivetti-style 'fancy)
    (olivetti-mode 1)))

(add-hook 'org-mode-hook #'thomas/olivetti-if-chapter)

(add-hook! 'org-mode-hook (display-line-numbers-mode -1))
